cmd/document-crds.go

```diff
@@ -156,7 +165,16 @@ func (options *documentCRDsOptions) validate(
 }
 
 // applyToConfig applies options we've received on the command line to the config.
-func (options *documentCRDsOptions) applyToConfig(cfg *config.Config) {
+func (options *documentCRDsOptions) applyToConfig(cmd *cobra.Command, cfg *config.Config) {
 	cfg.SetTemplatePath(options.templatePath)
-	cfg.EnableClassDiagrams(options.classDiagrams)
+	
+	// Only apply classDiagrams if the flag was explicitly set
+	if cmd.Flags().Changed("class-diagrams") {
+		cfg.EnableClassDiagrams(options.classDiagrams)
+	}
+	
+	// Only apply useGoFieldNames if the flag was explicitly set
+	if cmd.Flags().Changed("use-go-field-names") {
+		cfg.SetUseGoFieldNames(options.useGoFieldNames)
+	}
```

Comment:
This is the wrong place to handle the commandline flags - they should all be handled within `newDocumentCRDsCommand` with the values all captured on the `documentCRDsOptions` instance.

Remove this and handle the new commandline flag in the same way as the existing ones, including modification of `documentCRDsOptions.applyToConfig()` to apply the new flag to the active config.